### PR TITLE
Add docker

### PR DESCRIPTION
**Pull request by andrewshawcare**
This adds docker and usage instructions in the documentation.

### CODE REVIEW

**Comment by domoritz** [comment]
Thank you for the pr! I don't have much experience with docker. Could you explain what the benefits are and why you use docker compose instead of just docker? Thanks!

**Comment by andrewshawcare** [comment]
No worries! Docker is many things to many people, but for me it's a way of trying out new tools without having to install anything on my computer (besides docker and docker compose). Every project can be used just by running `docker-compose up` which:

1. Builds a docker "image" using a "Dockerfile" which runs everything required to setup and use the project in what ends up being published as an "image" (If you've used Vagrant, this is very similar).
2. Creates a docker "container" which is a running version of that image
3. Mounts volumes and exposes ports so development can occur on the container

The benefits are that it's very easy to get started (I didn't have npm or yarn installed, this saved me from having to install them on my machine) and if I break something I can just start over by recreating the docker image. I can also share projects with others knowing things will "just work" as long as they can run docker.

In any event, if docker isn't your thing or if you think the project is easy enough to use as-is, feel free to reject the PR, just figured I'd add it in case it served any benefit to others.

File: README.md [hunk 1]
```diff
@@ -18,6 +18,20 @@ $ yarn
 $ yarn start
 ```
 
+3. The local web server will be accessible from (http://localhost:8080)[http://localhost:8080].
+
+#### Docker
+
+If you'd like to use Docker, there's a Docker Compose setup that you can use:
```
Comment (domoritz) [inline README.md]: Add a link to docker compose docs?

Comment (andrewshawcare) [reply README.md]: Added a link to both docker and docker compose overview pages

**Comment by domoritz** [comment]
I have to play with this a bit more but this is super neat so I'll merge it for now. 

I just found this tutorial, which I found very informative: https://jdlm.info/articles/2016/03/06/lessons-building-node-app-docker.html .

**Comment by domoritz** [comment]
Hmm, I started with a clean node_modules and still get errors


```
> docker-compose up
Creating editor_editor_1 ... done
Attaching to editor_editor_1
editor_1  | yarn run v1.12.3
editor_1  | $ webpack-dev-server --mode development --hot
editor_1  |  10% building 1/1 modules 0 activeℹ ｢wds｣: Project is running at http://0.0.0.0:8080/
editor_1  | ℹ ｢wds｣: webpack output is served from /
editor_1  | ℹ ｢wds｣: Content not from webpack is served from /usr/src/app/public
 10% building 7/11 modules 4 active ...es/webpack-dev-server/client/overlay.jsℹ ｢atl｣: Using typescript@3.2.2 from typescript
editor_1  | ℹ ｢atl｣: Using tsconfig.json from /usr/src/app/tsconfig.json (in a forked process)
 94% after sealℹ ｢atl｣: Checking started in a separate process...
✖ ｢wdm｣:
editor_1  | ERROR in ./src/index.tsx
editor_1  | Module not found: Error: Can't resolve 'vega-lite' in '/usr/src/app/src'
editor_1  |  @ ./src/index.tsx 7:0-32 13:23-25 15:38-48
editor_1  |  @ multi (webpack)-dev-server/client?http://0.0.0.0:8080 (webpack)/hot/dev-server.js ./src/index.tsx
editor_1  |
editor_1  | ERROR in ./src/components/app.tsx
editor_1  | Module not found: Error: Can't resolve 'vega-lite' in '/usr/src/app/src/components'
editor_1  |  @ ./src/components/app.tsx 9:0-33 31:16-20
editor_1  |  @ ./src/components/app-shell.tsx
editor_1  |  @ ./src/index.tsx
editor_1  |  @ multi (webpack)-dev-server/client?http://0.0.0.0:8080 (webpack)/hot/dev-server.js ./src/index.tsx
editor_1  |
editor_1  | ERROR in ./src/components/toolbar/renderer.tsx
editor_1  | Module not found: Error: Can't resolve 'vega-lite' in '/usr/src/app/src/components/toolbar'
editor_1  |  @ ./src/components/toolbar/renderer.tsx 4:0-32 7:47-57
editor_1  |  @ ./src/components/toolbar/index.tsx
editor_1  |  @ ./src/components/viz-pane/renderer.tsx
editor_1  |  @ ./src/components/viz-pane/index.tsx
editor_1  |  @ ./src/components/app.tsx
editor_1  |  @ ./src/components/app-shell.tsx
editor_1  |  @ ./src/index.tsx
editor_1  |  @ multi (webpack)-dev-server/client?http://0.0.0.0:8080 (webpack)/hot/dev-server.js ./src/index.tsx
editor_1  |
editor_1  | ERROR in ./src/reducers/index.ts
editor_1  | Module not found: Error: Can't resolve 'vega-lite' in '/usr/src/app/src/reducers'
editor_1  |  @ ./src/reducers/index.ts 1:0-32 24:48-58
editor_1  |  @ ./src/store/configure-store.ts
editor_1  |  @ ./src/index.tsx
editor_1  |  @ multi (webpack)-dev-server/client?http://0.0.0.0:8080 (webpack)/hot/dev-server.js ./src/index.tsx
editor_1  |
editor_1  | ERROR in ./src/components/app.tsx
editor_1  | Module not found: Error: Can't resolve 'vega-lite/build/src/util' in '/usr/src/app/src/components'
editor_1  |  @ ./src/components/app.tsx 10:0-48 54:12-16 54:46-50
editor_1  |  @ ./src/components/app-shell.tsx
editor_1  |  @ ./src/index.tsx
editor_1  |  @ multi (webpack)-dev-server/client?http://0.0.0.0:8080 (webpack)/hot/dev-server.js ./src/index.tsx
editor_1  |
editor_1  | ERROR in ./src/components/input-panel/spec-editor/renderer.tsx
editor_1  | Module not found: Error: Can't resolve 'vega-lite/build/vega-lite-schema.json' in '/usr/src/app/src/components/input-panel/spec-editor'
editor_1  |  @ ./src/components/input-panel/spec-editor/renderer.tsx 10:23-71
editor_1  |  @ ./src/components/input-panel/spec-editor/index.tsx
editor_1  |  @ ./src/components/input-panel/index.tsx
editor_1  |  @ ./src/components/app.tsx
editor_1  |  @ ./src/components/app-shell.tsx
editor_1  |  @ ./src/index.tsx
editor_1  |  @ multi (webpack)-dev-server/client?http://0.0.0.0:8080 (webpack)/hot/dev-server.js ./src/index.tsx
editor_1  |
editor_1  | ERROR in ./src/utils/validate.ts
editor_1  | Module not found: Error: Can't resolve 'vega-lite/build/vega-lite-schema.json' in '/usr/src/app/src/utils'
editor_1  |  @ ./src/utils/validate.ts 10:38-86
editor_1  |  @ ./src/reducers/index.ts
editor_1  |  @ ./src/store/configure-store.ts
editor_1  |  @ ./src/index.tsx
editor_1  |  @ multi (webpack)-dev-server/client?http://0.0.0.0:8080 (webpack)/hot/dev-server.js ./src/index.tsx
editor_1  | ℹ ｢wdm｣: Failed to compile.
editor_1  | ✖ ｢atl｣: Checking finished with 5 errors
editor_1  | [at-loader] ./src/components/toolbar/renderer.tsx:5:21
editor_1  |     TS2307: Cannot find module 'vega-lite'.
editor_1  |
editor_1  | [at-loader] ./src/components/app.tsx:9:22
editor_1  |     TS2307: Cannot find module 'vega-lite'.
editor_1  |
editor_1  | [at-loader] ./src/components/app.tsx:10:22
editor_1  |     TS2307: Cannot find module 'vega-lite/build/src/util'.
editor_1  |
editor_1  | [at-loader] ./src/reducers/index.ts:1:21
editor_1  |     TS2307: Cannot find module 'vega-lite'.
editor_1  |
editor_1  | [at-loader] ./src/index.tsx:7:21
editor_1  |     TS2307: Cannot find module 'vega-lite'.
editor_1  |
```

**Comment by andrewshawcare** [comment]
Interesting, is that from a clean checkout, or are there additional local changes? If you add the following volume to `docker-compose.yml`:

```
- "./src:/usr/src/app/src"
```

you should have the ability to edit source locally and have it sync to the container. If you add any new packages you'd have to rebuild the image. Let me know if you have some time tomorrow, I can pair with you to see if we can debug together.

**Pull request merged**